marlowe-playground-client/src/Projects.purs

```diff
@@ -0,0 +1,112 @@
+module Projects where
+
+import Control.Monad.Except (runExceptT)
+import Control.Monad.Reader (runReaderT)
+import Data.Array (filter)
+import Data.Bifunctor (lmap)
+import Data.DateTime.ISO as ISO
+import Data.Either (Either(..))
+import Data.Formatter.DateTime (FormatterCommand(..), format)
+import Data.Lens (assign, to, view, (^.))
+import Data.List (fromFoldable)
+import Data.Maybe (Maybe(..))
+import Data.Newtype (unwrap)
+import Effect.Aff.Class (class MonadAff)
+import Gist (Gist, gistCreatedAt, gistDescription, gistId)
+import Halogen (ClassName(..), ComponentHTML, HalogenM)
+import Halogen.Classes (flex)
+import Halogen.HTML (a, div, h1_, hr_, span, table, tbody, td, td_, text, th_, thead, tr, tr_)
+import Halogen.HTML.Events (onClick)
+import Halogen.HTML.Properties (class_, classes)
+import Marlowe (SPParams_, getApiGists)
+import Marlowe.Gists (playgroundGist)
+import Network.RemoteData (RemoteData(..))
+import Network.RemoteData as RemoteData
+import Prelude (Unit, Void, bind, const, discard, flip, map, mempty, pure, unit, ($), (<<<))
+import Projects.Types (Action(..), Lang(..), State, _projects)
+import Servant.PureScript.Ajax (errorToString)
+import Servant.PureScript.Settings (SPSettings_)
+import Text.Parsing.Parser (runParser)
+import Types (ChildSlots)
+
+handleAction ::
+  forall m.
+  MonadAff m =>
+  SPSettings_ SPParams_ ->
+  Action -> HalogenM State Action ChildSlots Void m Unit
+handleAction settings LoadProjects = do
+  assign _projects Loading
+  resp <- flip runReaderT settings $ runExceptT getApiGists
+  assign _projects $ lmap errorToString $ RemoteData.fromEither resp
+
+handleAction settings (LoadProject lang gistId) = pure unit
+
+render ::
+  forall m.
+  MonadAff m =>
+  State ->
+  ComponentHTML Action ChildSlots m
+render state =
+  div [ classes [ ClassName "projects-container" ] ]
+    [ h1_ [ text "Open Project" ]
+    , hr_
+    , body (view _projects state)
+    ]
+  where
+  body (Success gists) = gistsTable $ filter playgroundGist gists
+
+  body (Failure _) = span [ class_ (ClassName "error") ] [ text "Failed to load gists" ]
+
+  body Loading = span [ class_ (ClassName "loading") ] [ text "Loading..." ]
+
+  body NotAsked = text mempty
+
+gistsTable ::
+  forall m.
+  MonadAff m =>
+  Array Gist ->
+  ComponentHTML Action ChildSlots m
+gistsTable gists =
+  table [ classes [ ClassName "gists" ] ]
```

Comment:
I really think you want some styling for the `gistsTable []` case.

Reply:
It looks a bit broken as it stands:

<img width="1050" alt="Screenshot 2020-10-02 at 16 30 09" src="https://user-images.githubusercontent.com/380756/94944347-0a27f980-04d1-11eb-8346-a3227d3ac7ee.png">

Reply:
Actually, thinking about it I might put the handling for that at line 56.

Reply:
ok changed to `No saved projects found` however unstyled. All the style feedback will go to the designer but this PR can still be merged.